File: cs12b/pa3/DictionaryTest.java
//DictionaryTest.java
//Camille Rogers
//carogers
//pa3, 12B
//test client for the Dictionary ADT while it is under construction. 

public class DictionaryTest{

   public static void main(String[] args){
      Dictionary A = new Dictionary();
      
      /*A.insert("1","cat"); //This tests my insert() method in Dictionary.java 
      A.insert("2","dog");
      A.insert("3","camille");
      A.insert("4","unicorn");
      A.insert("5","elephant");
      A.insert("6","frog");
      A.insert("7","goldfish");
      System.out.println(A.toString());*/
      
      /*String v;
      v = A.lookup("1");
      System.out.println("key=1 "+(v==null?"not found":("value="+v))); //This tests my lookup() method in Dictionary.java 
      v = A.lookup("4");
      System.out.println("key=4 "+(v==null?"not found":("value="+v)));
      v = A.lookup("7");
      System.out.println("key=7 "+(v==null?"not found":("value="+v)));
      v = A.lookup("6");
      System.out.println("key=6 "+(v==null?"not found":("value="+v)));
      System.out.println();*/

     /*A.delete("1"); //test my delete() in Dictionary.java 
      A.delete("3");
      A.delete("7");
      System.out.println(A.toString());*/
      
      System.out.println(A.isEmpty()); //checks if isEmpty() returns correct boolean and numItems
      System.out.println(A.size());
      A.makeEmpty();
      System.out.println(A.isEmpty());
      System.out.println(A);
   
   }
}

File: cs12b/pa5/DictionaryTest.c
//DictionaryTest.c
//Camille Rogers
//carogers
//12B pa5
//Test client for the Dictionary ADT

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){
   Dictionary A = newDictionary(); //creates new dictionary 
   char* k;
   char* v;
   char* test1[] = {"1","2","3","4","5"};
   char* test2[] = {"cat","dog","pig","frog","turtle"};
   int i;

   for(i=0; i<5; i++){
      insert(A, test1[i], test2[i]); //test insert()
   }

   printDictionary(stdout, A); //test print

   for(i=0; i<5; i++){
      k = test1[i];
      v = lookup(A, k); //test lookup
      printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
   }

   // insert(A, "five", "rat"); // error: duplicate keys

   delete(A, "2"); //tests delete()
   delete(A, "4");

   printDictionary(stdout, A); //test print 

   for(i=0; i<5; i++){
      k = test1[i];
      v = lookup(A, k);
      printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
   }

   // delete(A, "1");  // error: key not found

   printf("%s\n", (isEmpty(A)?"true":"false"));
   printf("%d\n", size(A));
   makeEmpty(A); //tests makeEmpty()
   printf("%s\n", (isEmpty(A)?"true":"false"));

   freeDictionary(&A);

   return(EXIT_SUCCESS);
}